astropy/coordinates/representation.py

```diff
@@ -128,7 +128,7 @@ def new_like(self, reps, length, metadata_conflicts='warn', name=None):
 
         Returns
         -------
-        col : Representation or Differential
+        col : `BaseRepresentation` or `BaseDifferential` subclass instance
```

Comment:
```suggestion
        col : representation-instance or differential-instance
```
Similar to suggestion for frame-instance above - we could add these as terms? Same comment for many lines below, but good to decide on this before I update my suggestions below...